### ISSUE TITLE
bitnami thanos fails with err="read /conf/objstore.yml: is a directory..."

### ISSUE THREAD

**Issue by hrisant** [bitnami/charts#3395]
Hi,

I am trying to deploy the thanos chart with a simple config based on s3, without the minio chart(maybe this is the mistake?):

objstoreConfig:
  type: S3
  config:
    bucket: {{ thanos.s3_bucket.name }}
    endpoint: "s3.{{ terraform.region }}.amazonaws.com"
    insecure: false
    signature_version2: false
    encrypt_sse: true
    put_user_metadata:
      "X-Amz-Acl": "bucket-owner-full-control"
    http_config:
      idle_conn_timeout: 0s

image:
  registry: docker.io
  repository: bitnami/thanos
##tag: 0.14.0-scratch-r1
  tag: latest
querier:
.............................................................................

this is actually a j2 template that generates the yaml.
The yaml file is generated ok, the secret thanos-objstore-secret is generated ok with the intended content.
But all the pods except querier(bucketweb, compactor, ruler, storegateway) are in crashloop with following error:

level=error ts=2020-08-12T12:18:59.560080696Z caller=main.go:185 err="read /conf/objstore.yml: is a directory\nloading YAML file /conf/objstore.yml for objstore.config-file\ngithub.com/thanos-io/thanos/pkg/extflag.(*PathOrContent).Content\n\t/app/pkg/extflag/pathorcontent.go:61\nmain.runCompact\n\t/app/cmd/thanos/compact.go:160\nmain.registerCompact.func1\n\t/app/cmd/thanos/compact.go:84\nmain.main\n\t/app/cmd/thanos/main.go:183\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:203\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1373\npreparing compact command failed\nmain.main\n\t/app/cmd/thanos/main.go:185\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:203\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1373"

From this I understand that it's trying to read objstore configuration from the path /conf/objstore.yml but it finds a directory there instead of a file.
Looking in the compactor deployment for example(in master) I see:

.................................
volumeMounts:
            - name: objstore-config
              {{- if .Values.existingObjstoreSecretItems }}
              mountPath: /conf
              {{- else }}
              mountPath: /conf/objstore.yml
              {{- end }}
..................................
volumes:
        - name: objstore-config
          secret:
            secretName: {{ include "thanos.objstoreSecretName" . }}
            {{- if .Values.existingObjstoreSecretItems }}
            items: {{- toYaml .Values.existingObjstoreSecretItems | nindent 14 }}
            {{- end }}
.......................................

which I think it means that, without the  existingObjstoreSecretItems arg, the file objstore.yaml found in the kube secret will be mounted in container in the path /conf/objstore.yml/objstore.yml.
Am I missing something? I saw in fetched chart there is also a subpath setting there, not sure if has any influence on this.
I can provide the full generated yaml file if needed. 
Please advise!

All the best,
Andrei

**Comment by joancafom** [comment]
Hi @hrisant ,

Thanks for your issue! Some other users have already reported this flaw (see https://github.com/bitnami/charts/issues/3375) and there is an ongoing PR #3396 intended to solve this. One of our engineers will review it soon and eventually it will be merged 😁.

Thanks!

**Comment by hrisant** [comment]
Hi @joancafom ,

Thanks for the answer! I missed that opened issue. It's great that there is already a PR.
Waiting for it to merge with great anticipation:)

All the best,
Andrei

**Comment by hrisant** [comment]
Should I close this then?

**Comment by joancafom** [comment]
Hi @hrisant 

Thanks! You don't need to worry 😁, once the PR is merged the issue will be automatically closed.

Regards

**Comment by hrisant** [comment]
Thanks @joancafom!

**Comment by hrisant** [comment]
Hey @joancafom , 

This is kind of unrelated and I know this is not a support channel, but just in case you can point me quickly in the right direction:

if I have 3 prometheus shards, with 3 thanos-sidecars and 3 services for these sidecars of the form:
sidecar-edge-peers                         LoadBalancer
sidecar-kube-peers                         LoadBalancer
sidecar-infr-peers                         LoadBalancer

is there a way I can use the querier chart param "querier.dnsDiscovery.sidecarsService" for all three?
In this doc (https://docs.bitnami.com/tutorials/create-multi-cluster-monitoring-dashboard-thanos-grafana-prometheus/) I found this form:

querier:
  stores:
    - SIDECAR-SERVICE-IP-ADDRESS-1:10901
    - SIDECAR-SERVICE-IP-ADDRESS-2:10901

But I would rather not depend on IPs as we sometime redeploy our monitoring.
Again sorry for abusing this issue! Only if it is a quick answer, if not, I'll keep digging.
Thanks!

**Comment by joancafom** [comment]
Hi @hrisant 

Would you mind opening another issue for this? This issue will be closed as soon as the PR is merged, but we can move the discussion to the other ticket 😁

Thanks!

**Comment by hrisant** [comment]
Sure thing!
Thanks @joancafom !
